src/main/java/com/iexec/worker/pubsub/SubscriptionService.java

```diff
@@ -107,15 +114,47 @@ synchronized void reSubscribeToTopics() {
         Set<String> chainTaskIds = this.chainTaskIdToSubscription.keySet();
         if (chainTaskIds.isEmpty()) {
             log.info("No topic to resubscribe to");
-            return;
+        } else {
+            log.info("ReSubscribing to topics [chainTaskIds: {}]", chainTaskIds);
+            chainTaskIds.forEach(chainTaskId -> {
+                this.chainTaskIdToSubscription.remove(chainTaskId);
+                subscribeToTopic(chainTaskId);
+            });
+
+            log.info("ReSubscribed to topics [chainTaskIds: {}]", chainTaskIds);
         }
 
-        log.info("ReSubscribing to topics [chainTaskIds: {}]", chainTaskIds);
-        chainTaskIds.forEach(chainTaskId -> {
-            this.chainTaskIdToSubscription.remove(chainTaskId);
-            subscribeToTopic(chainTaskId);    
-        });
-        log.info("ReSubscribed to topics [chainTaskIds: {}]", chainTaskIds);
+        sessionReady = true;
```

Comment:
For 
`sessionReady = true;`
&
`sessionReady = false;`
should we warn if same state was already set?

Reply:
Right. This may help us detect issue in our flow. I'll add these logs.

Reply:
Please see https://github.com/iExecBlockchainComputing/iexec-worker/pull/497/commits/0ef14e7f28e06a3b029353be2d827c487a707a35.

Reply:
Thanks :ok_hand: